app/models/grda_warehouse/client_file.rb

```diff
@@ -46,6 +48,17 @@ def self.any_visible_by?(user)
       user.can_manage_window_client_files? || user.can_see_own_file_uploads?
     end
 
+    def set_client_consent
+      if consent_form_confirmed_changed? || consent_form_signed_on_changed?
+        date = consent_form_confirmed? ? consent_form_signed_on : nil
+        client.update_column :consent_form_signed_on, date
+      end
+    end
+
+    def clear_client_consent
```

Comment:
This looks super dangerous since we may have multiple consent forms and there doesn't appear to be a check to see if this is the newest one.   I think we can safely just leave whatever is on the client, or at least verify that the date on the client matches the date on the file before we remove it from the client.  In theory at least, the form had to be confirmed before it could get pushed to the client.

Reply:
Oops, yep, guess i was thinking there would only be one consent form at any time.

Reply:
oh man, i wasnt even checking if the file was of type 'Consent Form'. Doh!